app/src/main/java/com/github/warnastrophy/core/data/repository/NominatimRepository.kt

```diff
@@ -120,33 +118,27 @@ class NominatimRepository(private val ioDispatcher: CoroutineDispatcher = Dispat
    * @param urlStr The full URL to fetch.
    * @return The response body as a string, or an empty string if rate limited.
    */
-  private suspend fun httpGet(urlStr: String): String =
-      withContext(ioDispatcher) {
-        val bool = isRateLimited()
-
-        if (bool) {
-          return@withContext ""
-        }
-
-        val url = URL(urlStr)
-        val conn =
-            (url.openConnection() as HttpURLConnection).apply {
-              requestMethod = "GET"
-              setRequestProperty("Accept-Language", "en")
-              setRequestProperty("User-Agent", "WarnAStrophyApp/1.0 (+${referer}")
-              setRequestProperty("Referer", referer)
-            }
-        val message: String
-        try {
-          val code = conn.responseCode
-          val stream = if (code in 200..299) conn.inputStream else conn.errorStream
-          message = BufferedReader(InputStreamReader(stream)).use { it.readText() }
-        } finally {
-          conn.disconnect()
+  private fun httpGet(urlStr: String): String {
+    lastQueryTimestamp = System.currentTimeMillis()
```

Comment:
You changed `httpGet` from `suspend` + `withContext` to a blocking function. If other callers expect `suspend` or `withContext` behavior this can introduce blocking IO on the wrong thread. You need to ensure all calls happen on IO dispatcher. However it seems to not be the case currently in `reverseGeocode`. This is why I suggest taking back the `suspend` + `withContext.`

```suggestion
  private suspend fun httpGet(urlStr: String): String =
      withContext(ioDispatcher) {
        lastQueryTimestamp = System.currentTimeMillis()
```

Reply:
- `suspend` enables pausing and resuming, but **does not itself manage threads**.
- `withContext(Dispatchers.IO)` explicitly switches to a background thread for blocking I/O.

Okay for the `suspend` but, because the first author of httpGet chose to follow caller convention for the thread managing, you have to call `httpGet` in a coroutine scope with runs on `Dispatchers.IO`.

Reply:
I agree. If you look at nominatim Service you will notice that It takes in argument a Dispatcher which is initialized to Dispatchers.IO. Reversegeocode is called in this dispatcher which is in fact an IO one